--- a/CMakeLists.txt	2017-10-14 23:18:12.000000000 +0900
+++ b/CMakeLists.txt	2019-05-14 16:00:48.587045481 +0900
@@ -22,10 +22,7 @@
 include (GetGitRevisionDescription)
 git_describe (VERSION "--tags")
 
-if ("x_${VERSION}" STREQUAL "x_GIT-NOTFOUND" OR "x_${VERSION}" STREQUAL "x_HEAD-HASH-NOTFOUND" OR "x_${VERSION}" STREQUAL "x_-128-NOTFOUND")
-  message (WARNING " - Install git to compile a production libtinyb!")
-  set (VERSION "v0.5.0-dirty")
-endif ()
+  set (VERSION "v0.5.1")
 
 message (INFO " - libtinyb Version ${VERSION}")
 
